MapboxDirections/MBRouteOptions.swift

```diff
@@ -160,7 +160,6 @@ open class RouteOptions: NSObject, NSSecureCoding, NSCopying{
      */
     @objc public init(waypoints: [Waypoint], profileIdentifier: MBDirectionsProfileIdentifier? = nil) {
         assert(waypoints.count >= 2, "A route requires at least a source and destination.")
-        assert(waypoints.count <= 25, "A route may not have more than 25 waypoints.")
```

Comment:
This assertion throws for map matching now. Can we let the server handle this error?

Reply:
This is a strong sign that RouteOptions and MatchOptions need to be siblings under either an abstract superclass or a protocol. For now, though, let’s factor the assertions out into a private preflight method that gets called from this initializer. The MatchOptions initializer can override that method with the assertions that make sense for that API.